### ISSUE TITLE
Unrealistic Accuracy? Examining Ground Truth Leakage in ImageNet Testing

### ISSUE THREAD

**Issue by dangai2021** [nick-nikzad/SATA#2]
It could be that the ground truth labels are unintentionally leaking into the evaluation. Otherwise, it’s unclear how the model achieves over 93% Top-1 accuracy as reported in the paper.

Option 1: Testing with shuffle=False in the validation dataset
dataloader = torch.utils.data.DataLoader(imagenet_dataset, batch_size=args.batch_size, shuffle=False, num_workers=4)

Result:
Final Top-1 Accuracy: 93.78%
Final Top-5 Accuracy: 99.77%

Option 2: Testing with shuffle=True in the validation dataset
dataloader = torch.utils.data.DataLoader(imagenet_dataset, batch_size=args.batch_size, shuffle=True, num_workers=4)

Result:
Final Top-1 Accuracy: 53.64%
Final Top-5 Accuracy: 79.81%

**Comment by nick-nikzad** [comment]
@SnowXYZ2021 First of all, a delay in response does not mean we are avoiding you. If there is urgency, please feel free to reach out through formal channels. Otherwise, due to workload, we may not be able to respond promptly, and checking here frequently is impractical.

Secondly, since the proposed SATA does not involve any training and all evaluation is performed on the validation set, there is no risk of data leakage between the training and validation sets. It is also a convention to set shuffle to False for validation. Even with this assumption, our findings remain valid with no leakage or such accusation you refer to. Using shuffle=True would be a matter for future investigation, which could be valuable for researchers in this domain.

**Comment by zeyuanyin** [comment]
The method achieves 95% top-1 accuracy on the ImageNet-1K validation set, but drops to 50% after shuffling. This suggests it may be exploiting the default ordering of the dataset—where samples are sorted by class (e.g., the first 50 images are class 0, the next 50 are class 1, and so on)—rather than truly learning from image content.

Shuffling is a way to break this default ordering, and the performance drops a lot.

Also, we can only change the batchsize to a very small value, such as8, and the performance drops to ~77%.

**Comment by SnowXYZ2021** [comment]
@nick-nikzad The author's response is underwhelming, suggesting a potential lack of foundational knowledge in deep learning. Relying on shuffle to enhance performance is a novice error, I still harbor suspicions that the author is intentionally perpetrating academic dishonesty. If the author genuinely believes their approach is valid, I would like to pose a pointed question: what is the practical significance of such unstable inference? For example, when identifying a cat, recognizing a single cat image versus bundling multiple animals in a batch for recognition can produce disparate results. This is utterly implausible🤣

**Comment by nick-nikzad** [comment]
@SnowXYZ2021 The fact is that everything we have done is transparent and open to the world, unlike hiding behind fake IDs. No research work is ever completely flawless, and we have never claimed otherwise. However, we demonstrated that with just simple calculations and analysis, it is possible to achieve significant performance improvements—potentially opening the door for others to build upon and complete the idea.

**Comment by zeyuanyin** [comment]
> [@nick-nikzad](https://github.com/nick-nikzad) Receiving criticism on a top-tier CV conference paper is a common occurrence, irrespective of the source, even if it comes from anonymous IDs. Relying on this as a defense mechanism lacks persuasiveness. You can consider me as an anonymous reviewer evaluating your paper through a single-blind review process.
> 
> Addressing the concerns raised earlier, it is inappropriate and disgraceful to manipulate the order or naming conventions of images to enhance performance. It is imperative that you present the prediction results for each image using a batch size of 1.
> 
> Moreover, it is confusing that the author proposes, "using shuffle=True warrants further investigation, which could be beneficial for researchers in this field." This assertion is not only implausible but also nonsensical. I strongly discourage disseminating such misinformation within the academic community.😅

I did the experiment with batch size of 1 without any other changes. The final result is 80.64%,  which is even lower than off-the-shelf deit-b's 82%. 

<img width="1202" height="197" alt="Image" src="https://github.com/user-attachments/assets/5e8f8a4d-8c20-4724-883d-a473faf64b0b" />
